### PR TITLE
Add bibliography submodule

### PR DESCRIPTION
**Pull request by MakisH**
Description of changes:
- add [the-teachingRSE-project/bibliography](https://github.com/the-teachingRSE-project/bibliography) as a submodule of this project
- update the contributing guidelines
- remove the local BibTeX file
   - migrate any contributions from currently open PRs to the new BibTeX repository
   - closes #245

### CODE REVIEW

**Comment by jngrad** [comment]
> * Adds a workflow to update the submodules (manually/on a schedule)

I have reservations about automatically updating the submodule revision: any commit in the BibTeX repository would trigger a new commit in N repositories. If the commit contains a syntax error, CI will break in N repositories. At my institute, we introduced CI in the BibTeX repository to mitigate that issue, and required the update to be done manually, so that the individual updating the submodule is also responsible for checking the entire bibliography in the manuscript.

**Comment by MakisH** [comment]
I can make the workflow manual, that's not a problem.
We definitely don't want a cascade of commits and errors.

File: Makefile [hunk 1]
```diff
@@ -6,12 +6,12 @@ objects := competencies.pdf
 
 all: $(objects)
 
-%.pdf: %.md bibliography.bib contributors.yml preamble.sty build/template.tex glossary.tex filter.py
+%.pdf: %.md bibliography/bibliography.bib contributors.yml preamble.sty build/template.tex glossary.tex filter.py
```
Comment (MakisH) [inline Makefile:9]: @jngrad any clue why the CI workflow complains that there is no target `competencies.pdf`?

_It works on my machine_ ©️

Comment (jngrad) [reply Makefile:9]: The GitHub Action was missing the submodule checkout, hence the `bibliography/bibliography.bib` requirement of the Makefile recipe was missing.

**Review by jngrad — APPROVED** [review]
LGTM

File: Makefile (hunk 1 above)
Comment (MakisH) [reply Makefile:9]: of course... :see_no_evil: 
Thanks for the quick fix!

**Comment by jngrad** [comment]
Everything works as expected: https://github.com/the-teachingRSE-project/institutionalized_education/pull/4 and compile action [9319020130](https://github.com/jngrad/RSE_institutionalized_education/actions/runs/9319020130).

**Comment by MakisH** [comment]
From my side, this is ready to merge. I will not join the next two meetings, please go ahead and implement any edits that might pop up.

**Review by CaptainSifff — APPROVED** [review]
I agree, technically this looks great!
I now wonder, what the workflow is for MRs like the changes to 266 by @hvwaldow ...

**Pull request merged**